### PR TITLE
Consolidate the directives index

### PR DESCRIPTION
**Pull request by raboof**

### CODE REVIEW

File: docs/src/main/paradox/scala/http/routing-dsl/directives/index.md [hunk 1]
```diff
@@ -116,18 +138,60 @@ is a lot more concise and as such more readable and maintainable. In addition it
...
 
+@@@
+
+@@@ div { .group-java }
+
+Writing multiple routes that are tried as alternatives (in-order of definition), is as simple as using the `route(route1, route2)`,
+method:
+
+```java
+Route routes = route(
```
Comment (jlprat) [inline docs/src/main/paradox/scala/http/routing-dsl/directives/index.md]: Does it make sense to move these to a `.java` file?

File: docs/src/main/paradox/scala/http/routing-dsl/directives/index.md [hunk 2]
```diff
@@ -146,31 +210,56 @@ transformations, both (or either) on the request and on the response side.
...
 Let's take a look at this concrete example:
 
-@@snip [DirectiveExamplesSpec.scala]($test$/scala/docs/http/scaladsl/server/DirectiveExamplesSpec.scala) { #example-1 }
+Scala
+:  @@snip [DirectiveExamplesSpec.scala]($test$/scala/docs/http/scaladsl/server/DirectiveExamplesSpec.scala) { #example-1 }
+
+Java
+:  @@snip [DirectiveExamplesTest.java]($test$/java/docs/http/javadsl/server/DirectiveExamplesTest.java) { #example1 }
+
+Here the `get` and `put` directives are chained together @scala[with the `~` operator]@java[using the `orElse` method] to form a higher-level route that
```
Comment (ktoso) [inline docs/src/main/paradox/scala/http/routing-dsl/directives/index.md]: eventually start recommending `concat()` for scala I guess. But don't mind right now, let's just get them all consolidated first

File: docs/src/test/scala/docs/http/scaladsl/server/DirectiveExamplesSpec.scala [hunk 3]
```diff
@@ -30,7 +30,7 @@ class DirectiveExamplesSpec extends RoutingSpec {
   }
 
   "example-2" in {
-    //#example-2
+    //#getOrPut
```
Comment (ktoso) [inline docs/src/test/scala/docs/http/scaladsl/server/DirectiveExamplesSpec.scala]: 👍

**Comment by raboof** [comment]
Agreed, would be nice to do a further cleanup at some point, but for now let's at least get to one tree :)

**Pull request merged**